xDiffCollection/CollectionBin.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  CollectionBin.swift
+//  CollectionTest
+//
+//  Created by Esteban Garro on 2018-06-05.
+//  Copyright © 2018 aldo. All rights reserved.
+//
+
+import Foundation
+
+internal enum BinOperationType {
+    case updated
+    case added
+    case deleted
+    case none
+}
+
+internal struct BinResult {
+    var type: BinOperationType
+    var idx : Int
+}
+
+internal protocol CollectionFiltering {
+    func filter<T:Hashable>(_ element:T) -> Bool
+}
+
+internal struct CollectionBin<T:Hashable> : CollectionFiltering, CustomStringConvertible {
+    var index : Int
+    var elements:Array<T> = []
+    var filter:(T)->Bool
+    var description: String {
+        var resp = ""
+        for (idx, element) in self.elements.enumerated() {
+            resp += "\(idx))\t\(element.hashValue)\n"
+        }
+        return resp
+    }
+    var count : Int {
+        return elements.count
+    }
+    
+    init(index: Int, filter: @escaping (T)->Bool, elements: Array<T>) {
```

Comment:
I guess filter should be the last because it's closure

Reply:
Ooops, yes.